### PR TITLE
ZIO Test: migrate SinkSpec

### PR DESCRIPTION
**Pull request by regiskuckaertz**
It's all there except for the `utf8Decode` which are still failing, I think because the generated string has a leading `0x00` whereas the decoded one doesn't. Any idea why @iravid ?

### CODE REVIEW

**Comment by iravid** [comment]
@regiskuckaertz Could be a bug, can you create a specific test for that case so I can see?

**Comment by iravid** [comment]
Oh, in @msosnicki reported in #1831 that `utf8Decode` is broken. Let's just delete it @regiskuckaertz.

**Comment by regiskuckaertz** [comment]
@iravid Both `utf8Decode` and `utf8DecodeChunk`?

**Comment by iravid** [comment]
No, just `utf8Decode`. The chunk version should be sound

**Comment by regiskuckaertz** [comment]
Hmm ok, it fails too then I'll try to provide a minimal test case.

**Comment by iravid** [comment]
Ah ok. Yes a test case would be good. Can you also delete `utf8Decode` itself?

**Comment by regiskuckaertz** [comment]
Hmm, you're right. It may be the `anyChar` generator that does not produce valid characters? I have updated the test to compare byte arrays instead, using `.getBytes("UTF-8")` and it passes with flying colours.

**Comment by regiskuckaertz** [comment]
woops, flaky "supervise fibers in fork"

**Comment by regiskuckaertz** [comment]
Interesting conundrum: 2.11 won't compile this line if I don't use `res.right.map`, but 2.13 won't if I use it 🤔 I thought the build had some kind of shim for that.

**Comment by iravid** [comment]
@regiskuckaertz just pattern match

**Comment by vasilmkd** [comment]
Maybe `streams` or `streamsTests` is not properly configured to pick up the version specific shims?

File: test/shared/src/main/scala/zio/test/Gen.scala [hunk 1]
```diff
@@ -112,6 +112,24 @@ object Gen extends GenZIO with FunctionVariants {
         .map(Sample.shrinkIntegral(0))
     }
 
+  /**
+   * A generator of Unicode characters. Shrinks toward '0'.
+   * Heavily inspired from:
+   * https://github.com/typelevel/scalacheck/blob/ab15a9fe012ca7c9feb48ee188e4d167df76d6ba/src/main/scala/org/scalacheck/Arbitrary.scala#L122-L134
+   */
+  final val anyUnicodeChar: Gen[Random, Char] = {
```
Comment (regiskuckaertz) [inline test/shared/src/main/scala/zio/test/Gen.scala]: ping @adamgfraser I'll open a separate PR for this

**Comment by regiskuckaertz** [comment]
Boom, removed.

**Comment by regiskuckaertz** [comment]
Another case of tests running fine locally but timing out in CircleCI 🤔 @softinio can you help me look into this?

File: streams-tests/jvm/src/test/scala/zio/stream/SinkSpec.scala [hunk 2]
```diff
@@ -488,43 +488,43 @@ object SinkSpec
           suite("orElse")(
             testM("left") {
               val sink = ZSink.identity[Int] orElse ZSink.fail("Ouch")
-              assertM(sinkIteration(sink, 1), equalTo((Left(1): Either[Int, String], Chunk.empty: Chunk[Int])))
+              assertM(sinkIteration(sink, 1), equalTo((Left(1): Either[Int, String], Chunk.empty)))
```
[the remark this answers is not in the document]
Comment (regiskuckaertz) [reply streams-tests/jvm/src/test/scala/zio/stream/SinkSpec.scala]: Done, good catch 👌

**Comment by regiskuckaertz** [comment]
@iravid ready now for review

**Comment by regiskuckaertz** [comment]
@adamgfraser i don't know why I didn't directly test that, but that was the stream generator using medium sized random 😌

**Comment by adamgfraser** [comment]
Nice. 😃 Well problem solved at least.

**Review by iravid — APPROVED** [review]
Amazing work Regis! And excellent detective work too :-)

**Pull request merged**